Classes/Issues/IssuesViewController.swift

```diff
@@ -352,6 +363,42 @@ IssueCommentSectionControllerDelegate {
             }
         }
     }
+    
+    func setLocked(_ locked: Bool) {
+        guard let currentStatus = current?.status else { return }
+        
+        let localModel = IssueStatusModel(
+            status: currentStatus.status,
+            pullRequest: currentStatus.pullRequest,
+            locked: locked
+        )
+        let localEvent = IssueStatusEventModel(
+            id: UUID().uuidString,
+            actor: client.sessionManager.focusedUserSession?.username ?? Strings.unknown,
+            commitHash: nil,
+            date: Date(),
+            status: locked ? .locked : .unlocked,
+            pullRequest: currentStatus.pullRequest
+        )
+        
+        localStatusChange = (localModel, localEvent)
+        feed.adapter.performUpdates(animated: true)
+        
+        client.setLocked(
+            owner: model.owner,
+            repo: model.repo,
+            number: model.number,
+            locked: locked
+        ) { [weak self] result in
+            switch result {
+            case .error:
+                self?.localStatusChange = nil
+                self?.feed.adapter.performUpdates(animated: true)
+                StatusBar.showGenericError()
+            default: break // dont need to handle success since updated optimistically
```

Comment:
Any reason to use `default:` over `.success:` and break there?

Reply:
Carbon copy of the updateStatus method above it with a few minor tweaks for locking - Didn't rewrite what was already working